src/cli/Generate.cpp

```diff
@@ -0,0 +1,110 @@
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <cstdlib>
+#include <stdio.h>
+
+#include "Generate.h"
+
+#include <QCommandLineParser>
+#include <QTextStream>
+
+#include "core/PasswordGenerator.h"
+
+Generate::Generate()
+{
+    this->name = QString("generate");
+    this->description = QObject::tr("Generate a new random password.");
+}
+
+Generate::~Generate()
+{
+}
+
+int Generate::execute(QStringList arguments)
+{
+    QTextStream inputTextStream(stdin, QIODevice::ReadOnly);
+    QTextStream outputTextStream(stdout, QIODevice::WriteOnly);
+
+    QCommandLineParser parser;
+    parser.setApplicationDescription(this->description);
+    QCommandLineOption len(QStringList() << "L" << "length",
+                                QObject::tr("Length of the generated password."),
+                                QObject::tr("length"));
+    parser.addOption(len);
+    QCommandLineOption lower(QStringList() << "l",
+                               QObject::tr("Use lowercase in the generated password."));
+    parser.addOption(lower);
+    QCommandLineOption upper(QStringList() << "u",
+                               QObject::tr("Use uppercase in the generated password."));
+    parser.addOption(upper);
+    QCommandLineOption numeric(QStringList() << "n",
+                               QObject::tr("Use numbers in the generated password."));
+    parser.addOption(numeric);
+    QCommandLineOption special(QStringList() << "s",
+                               QObject::tr("Use special characters in the generated password."));
+    parser.addOption(special);
+    QCommandLineOption extended(QStringList() << "e",
+                               QObject::tr("Use extended ascii in the generated password."));
+    parser.addOption(extended);
+    parser.process(arguments);
+
+    const QStringList args = parser.positionalArguments();
+    if (args.size() != 0) {
+        outputTextStream << parser.helpText().replace("keepassxc-cli", "keepassxc-cli generate");
+        return EXIT_FAILURE;
+    }
+
+    PasswordGenerator passwordGenerator;
+
+    if (parser.value(len).isEmpty()) {
+        passwordGenerator.setLength(PasswordGenerator::DefaultLength);
+    } else {
+        int length = parser.value(len).toInt();
+        passwordGenerator.setLength(length);
+    }
+
+    PasswordGenerator::CharClasses classes = 0x0;
+
+    if (parser.isSet(lower)) {
+        classes |= PasswordGenerator::LowerLetters; 
+    }
+    if (parser.isSet(upper)) {
+        classes |= PasswordGenerator::UpperLetters; 
+    }
+    if (parser.isSet(numeric)) {
+        classes |= PasswordGenerator::Numbers; 
+    }
+    if (parser.isSet(special)) {
+        classes |= PasswordGenerator::SpecialCharacters; 
+    }
+    if (parser.isSet(extended)) {
+        classes |= PasswordGenerator::EASCII; 
+    }
+
+    passwordGenerator.setCharClasses(classes);
```

Comment:
@TheZ3ro shouldn't we also set the default flags? If so we should do it in `edit` and `add` also.